File: backend/src/models/MedicalReport.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IMedicalReport extends Document {
    patientId: string;
    date: string; // ISO string
    fileName: string;
    fileUrl: string;
    doctorName: string;
    remarks?: string;
    createdAt: Date;
    updatedAt: Date;
}

const MedicalReportSchema: Schema = new Schema({
    patientId: { type: String, required: true, index: true },
    date: { type: String, required: true },
    fileName: { type: String, required: true },
    fileUrl: { type: String, required: true },
    doctorName: { type: String, required: true },
    remarks: { type: String }
}, {
    timestamps: true
});

export default mongoose.model<IMedicalReport>('MedicalReport', MedicalReportSchema);

File: backend-spring/src/main/java/com/aalliswell/entity/MedicalReport.java
package com.aalliswell.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "medical_reports", indexes = @Index(name = "idx_reports_patient", columnList = "patient_id"))
public class MedicalReport extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private User patient;

    @Column(nullable = false)
    private String date;

    @Column(nullable = false)
    private String fileName;

    @Column(nullable = false)
    private String fileUrl;

    @Column(nullable = false)
    private String doctorName;

    @Column(columnDefinition = "TEXT")
    private String remarks;
}
